src/main/java/ir/sahab/dockercomposer/DockerComposeRunner.java

```diff
@@ -137,17 +132,19 @@ private Service parseServiceState(String serviceState, String name) {
         final int internalPort = 7;
         final String externalIp = "127.0.0.1";
 
-        int firstSpace = serviceState.indexOf(' ');
-        String id = serviceState.substring(0, firstSpace);
+        String[] serviceInfoParts = serviceInfo.split("#");
+        String id = serviceInfoParts[0];
 
-        Matcher matcher = portPattern.matcher(serviceState);
         Map<Integer, Integer> portMappings = new HashMap<>();
-        // Each match will be a mapping from some internal port to some external port.
-        // The first published port will be considered as the default port.
-        while (matcher.find()) {
-            int exPort = Integer.parseInt(matcher.group(externalPort));
-            int inPort = Integer.parseInt(matcher.group(internalPort));
-            portMappings.put(inPort, exPort);
+        if (serviceInfoParts.length == 2) {
```

Comment:
Remove this 'if' statement.
Instead throw IllegalArgument if after the split, if the length is not '2'